pkg/pillar/pubsub/socketdriver/driver.go

```diff
@@ -218,3 +228,54 @@ func (s *SocketDriver) fixedDirName(name string) string {
 func (s *SocketDriver) persistentDirName(name string) string {
 	return fmt.Sprintf("%s/status/%s", "/persist", name)
 }
+
+//getBuffer returns a buffer and a done func to call at defer.
+func getBuffer() (*[]byte, func()) {
+	buf := bufPool.Get().(*[]byte)
+	return buf, func() {
+		bufPool.Put(buf)
+	}
+}
+
+// check and waits till conn's fd is readable
+func connReadCheck(conn net.Conn) error {
```

Comment:
Can't this function just cast the conn to get the unix conn, and then do a recvfrom with a zero length buffer?